Feature 2027 grid stat wind direction

hsoh-u opened the pull request:
## Expected Differences ##

There are minor differences because of precision (different hardware and different compiler versions)

- [x] Do these changes introduce new tools, command line arguments, or configuration file options? **[No]**</br>
If **yes**, please describe:</br>

- [x] Do these changes modify the structure of existing or add new output data types (e.g. statistic line types or NetCDF variables)? **[No]**</br>
If **yes**, please describe:</br>

## Pull Request Testing ##

- [x] Describe testing already performed for these changes:</br>

  - Run unit test at kiowa and seneca
  - Compare unit test output (especially grid_stat)

The unit test outputs are at
 -  kiowa:/d1/personal/hsoh/git/pull_request/MET_feature_2027_grid_stat_wind_direction/unit_test_output/
 -  seneca:/d1/personal/hsoh/git/features/feature_2027_grid_stat_wind_direction/MET/unit_test_output/

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[No]**

- [x] Do these changes include sufficient testing updates? **[No]**

- [x] Will this PR result in changes to the test suite? **[Yes]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

The threshold for wind is changed to ">5.0". The output of grid_stat/grid_stat_GFS_FOURIER_240000L_20120410_000000V.stat is changed.

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the PR definition above.
- [ ] Ensure the PR title matches the feature or bugfix branch name.
- [ ] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

Commit:
#2027 Added a debug message

Commit:
#2027 ci-run-unit Changed wind thres

Commit:
Merge remote-tracking branch 'origin/develop' into feature_2027_grid_stat_wind_direction

Commit:
#2027 ci-run-unit Increased the precision to debug message

georgemccabe committed:
print end of script after error summary, ci-run-unit

Commit:
#2027 Increased the buffer and cheking the status of sprintf

Commit:
Merge branch 'feature_2027_grid_stat_wind_direction' of github.com:dtcenter/MET into feature_2027_grid_stat_wind_direction

JohnHalleyGotway commented on met/src/libcode/vx_statistics/met_stats.cc:
```diff
@@ -1569,6 +1570,12 @@ void VL1L2Info::calc_ncep_stats() {
 
    DIR_ABSERR   = fabs(DIR_ERR);
 
+   char buffer[1024];
```
@hsoh-u this looks like temporary code that you added to debug this issue. But we wouldn't want to actually include this in the release. Recommend removing.

JohnHalleyGotway reviewed (changes requested):
Howard, I'd recommend removing that debug code from met_stats.cc.

If you see some need to keep it, I'd recommend replacing the use of that static character buffer with a call to ConcatString::format() instead.

hsoh-u committed:
Update met_stats.cc

#2047 Removed the debug message

hsoh-u commented:
It's removed

JohnHalleyGotway committed:
Update met_stats.cc

Removing variable that was set but never used because SonarQube might complain about that, like Fortify does.

JohnHalleyGotway reviewed (approved):
Simple change. Looks good to me. Thanks Howard.

Note that this will change the output of the unit tests and we'll need to merge develop into develop-ref after merging this into develop.

All the unit tests ran, but as expected, differences were flagged in the output. The diffs are limited to vector partial sums and statistics in a single Grid-Stat output file, which is exactly what they should be based on the change to the Grid-Stat config file.

```
COMPARING grid_stat/grid_stat_GFS_FOURIER_240000L_20120410_000000V.stat
file1: /data/output/met_test_truth/grid_stat/grid_stat_GFS_FOURIER_240000L_20120410_000000V.stat
file2: /data/output/met_test_output/grid_stat/grid_stat_GFS_FOURIER_240000L_20120410_000000V.stat
```
